R/wdpa_fetch.R

```diff
@@ -157,17 +157,20 @@ wdpa_fetch <- function(x, wait = FALSE,
                        datatype = "gdb",
                        verbose = interactive()) {
   # check that arguments are valid
+  ## expand path to ensure full path to folder
+  assertthat::assert_that(
+    assertthat::is.string(download_dir),
+    assertthat::noNA(download_dir))
+  download_dir <- path.expand(download_dir)
   ## check that classes are correct
-  # rappdirs::user_data_dir() returns a directory starting with ~/ for Mac, which causes the unzip to fail. Here we expand to remove ~. Shouldn't make any diffferent to other OS's where the path is already expanded.
-  download_dir <- path.expand(download_dir) 
-  dir.create(download_dir, showWarnings = FALSE, recursive = TRUE)
   assertthat::assert_that(
     assertthat::is.string(x),
     assertthat::is.dir(download_dir),
     assertthat::is.flag(force_download),
     assertthat::is.flag(verbose),
     assertthat::is.flag(check_version),
     identical(x, "global") || assertthat::is.string(country_code(x)))
+  dir.create(download_dir, showWarnings = FALSE, recursive = TRUE)
```

Comment:
The directory is not created until after the presence is tested `assertthat::is.dir(download_dir)`

Reply:
Ah - thanks for catching that. I'll push another commit